src/linker/Linker/Annotations.cs

```diff
@@ -606,6 +606,13 @@ internal bool DoesMethodRequireUnreferencedCode (MethodDefinition method, [NotNu
 				TryGetLinkerAttribute (method.DeclaringType, out attribute))
 				return true;
 
+			MethodDefinition? owningMethod;
+			while (context.CompilerGeneratedState.TryGetOwningMethodForCompilerGeneratedMember (method, out owningMethod)) {
+				if (DoesMethodRequireUnreferencedCode (owningMethod, out attribute))
+					return true;
+				method = owningMethod;
+			}
```

Comment:
Aren't the while loop and recursion redundant here? Why don't we only include the recursion:
```C#
if (DoesMethodRequireUnreferencedCode (owningMethod, out attribute))
	return true;
```

or make the whole function a do...while loop like:
```C#
do 
{
	if (method.IsStaticConstructor ()) {
		attribute = null;
		return false;
	}
	if (TryGetLinkerAttribute (method, out attribute))
		return true;
	if ((method.IsStatic || method.IsConstructor) && method.DeclaringType is not null &&
		TryGetLinkerAttribute (method.DeclaringType, out attribute))
		return true;
} while (context.CompilerGeneratedState.TryGetOwningMethodForCompilerGeneratedMember (method, out method))
```

Reply:
Good catch, the recursion was doing unnecessary work. I took your suggestion and removed the recursion here and for the corresponding code in the analyzer.